modules/stochastic_tools/include/utils/GaussianProcess.h

```diff
@@ -137,6 +137,72 @@ class GaussianProcess
    */
   void standardizeData(RealEigenMatrix & data, bool keep_moments = false);
 
+
+    void sq_dist(const RealEigenMatrix &X1_in, RealEigenMatrix &D_out, const RealEigenMatrix &X2_in = RealEigenMatrix(0,0));
+
+  struct Settings {
+    Real l;
+    Real u;
+    struct {
+      Real g;
+      Real theta;
+    } alpha, beta;
+  };
+
+  struct Initial {
+    RealEigenMatrix theta;
+    Real g;
+    Real tau2;
+  };
+
+  struct Output {
+    RealEigenMatrix x;
+    RealEigenMatrix y;
+    int nmcmc;
+    Settings settings;
+    Initial initial;
+    RealEigenMatrix g;
+    RealEigenMatrix theta;
+    RealEigenMatrix tau2;
+    RealEigenMatrix ll;
+  };
+
+  struct SampleGResult {
+    Real g;
+    Real ll;
+  };
+
+  struct SampleThetaResult {
+    Real theta;
+    Real ll;
+    Real tau2;
+  };
+
+  struct InvDetResult {
+    RealEigenMatrix Mi;
+    Real ldet;
+  };
+
+  struct LogLResult {
+    Real logl;
+    Real tau2;
+  };
+
+  void logl(const RealEigenMatrix & out_vec, const RealEigenMatrix & x1, const RealEigenMatrix & x2, Real g, const RealEigenMatrix & theta, 
```

Comment:
Add description for this function. See below for example from one of my code files. You should do something similar.

/**
   * Compute the acquisition function values
   * @param acq The computed acquisition function values
   * @param gp_mean The provided GP mean values
   * @param gp_std The provided GP standard deviation values
   * @param test_inputs All the input values under which the GP has to be tested
   * @param train_inputs All the input values under which the GP has been trained
   * @param generic A generic parameter (can be output values under which the GP has been trained or threshold parameter under the U-Function etc.)
   */
  virtual void computeAcquisition(std::vector<Real> & acq,
                                  const std::vector<Real> & gp_mean,
                                  const std::vector<Real> & gp_std,
                                  const std::vector<std::vector<Real>> & test_inputs,
                                  const std::vector<std::vector<Real>> & train_inputs,
                                  const std::vector<Real> & generic) const = 0;